pkg/operator/apiservicecontroller/apiservice_controller.go

```diff
@@ -230,3 +255,93 @@ func (c *APIServiceController) eventHandler() cache.ResourceEventHandler {
 		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
 	}
 }
+
+// TODO: change the name
+func (c *APIServiceController) filterAPIServices() []*apiregistrationv1.APIService {
+
+	oauthOperatorPrecondition := c.oauthOperatorAvailableAndVersionMatch()
```

Comment:
This needs to be a function you can pass in.  Make it a `WithSharedResponsibility`

Reply:
why ?

Reply:
> why ?

This is going to move to library-go so we can re-use it in openshift and oauth server operators